src/status_im2/common/standard_authentication/standard_auth/authorize.cljs

```diff
@@ -12,33 +12,38 @@
   (rf/dispatch [:set-in [:profile/login :error] ""]))
 
 (defn authorize
-  [{:keys [on-enter-password biometric-auth? on-auth-success on-auth-fail on-close
+  [{:keys [biometric-auth? on-auth-success on-auth-fail on-close
            auth-button-label theme blur? auth-button-icon-left]}]
-  (biometric/get-supported-type
-   (fn [biometric-type]
-     (if (and biometric-auth? biometric-type)
-       (biometric/authenticate
-        {:reason     (i18n/label :t/biometric-auth-confirm-message)
-         :on-success (fn [response]
-                       (when on-auth-success (on-auth-success response))
-                       (log/info "response" response))
-         :on-fail    (fn [error]
-                       (log/error "Authentication Failed. Error:" error)
-                       (when on-auth-fail (on-auth-fail error))
-                       (rf/dispatch [:show-bottom-sheet
-                                     {:theme   theme
-                                      :shell?  blur?
-                                      :content (fn []
-                                                 [enter-password/view
-                                                  {:on-enter-password on-enter-password}])}]))})
-       (do
-         (reset-password)
-         (rf/dispatch [:show-bottom-sheet
-                       {:on-close on-close
-                        :theme    theme
-                        :shell?   blur?
-                        :content  (fn []
-                                    [enter-password/view
-                                     {:on-enter-password on-enter-password
-                                      :button-icon-left  auth-button-icon-left
-                                      :button-label      auth-button-label}])}]))))))
+  (let [password-login   (fn [{:keys [on-press-biometrics]}]
+                           (rf/dispatch [:show-bottom-sheet
+                                         {:on-close on-close
+                                          :theme    theme
+                                          :shell?   blur?
+                                          :content  (fn []
+                                                      [enter-password/view
+                                                       {:on-enter-password   on-auth-success
+                                                        :on-press-biometrics on-press-biometrics
+                                                        :button-icon-left    auth-button-icon-left
+                                                        :button-label        auth-button-label}])}]))
+        biometrics-login (fn [on-press-biometrics]
```

Comment:
moved this to a var for functional purposes. LMK if anyone has a cleaner approach. 
Basically this kicks off biometrics authentication, but in the event biometrics fails it will fallback to password login, however that password screen will have a button to retrigger biometrics authentication. So this function needs to pass itself to as an argument

Reply:
maybe I should leave a comment on it?

Reply:
Yes, I think a comment would be nice.

It seems the reason for `biometrics-login` to be recursive is that we want to redispatch the event `:show-bottom-sheet` if biometrics fail, I think that's because we are executing `on-close` at the beggining of the `biometrics-login` fn.

So, what if we execute the `on-close` only when the user cancels or the login is successful?

I'm not familiar with this part of the code or UI flow, so I'm just suggesting.